src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -677,6 +678,95 @@ size_t runtime::gpu::CUDAEmitter::build_elementwise_n_to_1(const GPURuntimeConte
     return primitive_index;
 }
 
+size_t runtime::gpu::CUDAEmitter::build_reduce_window_helper(const GPURuntimeContext* ctx,
+                                                             const std::vector<std::string>& dtypes,
+                                                             const Shape& input_shape,
+                                                             const Shape& output_shape,
```

Comment:
Consider switching to GPUShape so that we are not using 64 bit words on the GPU unnecessarily.

Reply:
32bit definitely will save some space and computation, one concern is do we have limitation on shape to 32bit?

Reply:
Yep! We will have that check with the GPUShape PR, see https://github.com/NervanaSystems/ngraph/pull/1031/files#diff-87aac37cceeb79e1d3bf95706a2a804eR82